src/shogun/preprocessor/FeatureSelection.cpp

```diff
@@ -82,13 +82,6 @@ FeatureSelection<ST>::~FeatureSelection()
 
 }
 
-
-template <class ST>
-void FeatureSelection<ST>::cleanup()
-{
-	m_subset->remove_all_subsets();
```

Comment:
what about this?

Reply:
The author already used ```remove_all_subsets()``` inside the transform method (which is the public interface). If we call it n number of times, it will reset m_subset that many times and start anew. I suppose the author only implemented cleanup() because it was purely virtual.

Reply:
I checked the code, seems ok